### PR TITLE
chore: update example templates to use  `display_name`

### LINKED ISSUE #6978: update examples to use display_name
We recently added name a `display_name` to rich parameters so that we can have a nice machine-readable name and also change the display name without breaking existing workspaces. 

Let's update our examples to use this. We should be sure to update the provider version in each template too.

### PR DESCRIPTION
**Pull request by matifali**
This resolves #6978 
I validated all the templates with `terraform validate`

### CODE REVIEW

File: examples/templates/docker-with-dotfiles/main.tf [hunk 1]
```diff
@@ -28,24 +28,26 @@ data "coder_workspace" "me" {
...
 
 data "coder_parameter" "docker_image" {
-  name        = "What Docker image would you like to use for your workspace?"
-  description = "The Docker image will be used to build your workspace."
-  default     = "codercom/enterprise-base:ubuntu"
-  icon        = "/icon/docker.png"
-  type        = "string"
-  mutable     = false
+  name         = "docker_image"
+  display_name = "What Docker image would you like to use for your workspace?"
```
Comment (mtojek) [inline examples/templates/docker-with-dotfiles/main.tf:32]: Shouldn't that be `Docker Image`?

File: examples/templates/docker-with-dotfiles/main.tf [hunk 2]
```diff
@@ -28,24 +28,26 @@ data "coder_workspace" "me" {
...
+  icon         = "/icon/docker.png"
+  type         = "string"
+  mutable      = false
 }
 
 data "coder_parameter" "dotfiles_uri" {
-  name        = "What dotfiles repo would you like to use for your workspace?"
-  description = <<-EOF
+  name         = "dotfiles_uri"
+  display_name = "What dotfiles repo would you like to use for your workspace?"
```
Comment (mtojek) [inline examples/templates/docker-with-dotfiles/main.tf:42]: Display name should match name, so `dotfile URI`. You can move the question to the description.

File: examples/templates/gcp-windows/main.tf [hunk 3]
```diff
@@ -20,11 +20,12 @@ variable "project_id" {
 }
 
 data "coder_parameter" "zone" {
-  name    = "What region should your workspace live in?"
-  type    = "string"
-  default = "us-central1-a"
-  icon    = "/emojis/1f30e.png"
-  mutable = false
+  name         = "zone"
+  display_name = "What region should your workspace live in?"
```
Comment (mtojek) [inline examples/templates/gcp-windows/main.tf:24]: Same

**Review by mtojek — COMMENTED** [review]
Thanks for PR! Most comments refer to using `display_name` as a description, and for that we have a separate property.

File: examples/templates/docker-with-dotfiles/main.tf (hunk 1 above)
Comment (matifali) [reply examples/templates/docker-with-dotfiles/main.tf:32]: What I did is,
1. Copy `name` to `display_name`
2. Set `name` to equivalent `hcl` name 
	e.g. for
	```hcl
	data "coder_parameter" "docker_image"
                                 |
  		                         v
                                name --> display_name
	```

File: examples/templates/gcp-linux/main.tf [hunk 4]
```diff
@@ -20,11 +20,12 @@ variable "project_id" {
 }
 
 data "coder_parameter" "zone" {
-  name    = "What region should your workspace live in?"
-  type    = "string"
-  icon    = "/emojis/1f30e.png"
-  default = "us-central1-a"
-  mutable = false
+  name         = "zone"
+  display_name = "What region should your workspace live in?"
```
[the remark this answers is not in the document]
Comment (matifali) [reply examples/templates/gcp-linux/main.tf:24]: I will update as per the suggestion I followed 
```hcl
data "coder_parameter" "docker_image" 
                           | 
                           v 
                          name --> display_name
```

File: examples/templates/docker-with-dotfiles/main.tf (hunk 1 above)
Comment (mtojek) [reply examples/templates/docker-with-dotfiles/main.tf:32]: I see, but the overall idea is to use `display_name` for human-readable parameters.

**Comment by matifali** [comment]
@mtojek I have updated as per suggestions and moved the longer `display_name` fields to `description` where applicable

File: examples/templates/aws-windows/main.tf [hunk 5]
```diff
@@ -106,10 +107,11 @@ data "coder_parameter" "region" {
 }
 
 data "coder_parameter" "instance_type" {
-  name        = "Instance Type"
-  description = "What instance type should your workspace use?"
-  default     = "t3.micro"
-  mutable     = false
+  name         = "instance_type"
+  display_name = "Instance Type"
```
Comment (bpmct) [inline examples/templates/aws-windows/main.tf:111]:
```suggestion
  display_name = "Instance type"
```

File: examples/templates/aws-linux/main.tf [hunk 6]
```diff
@@ -106,10 +107,11 @@ data "coder_parameter" "region" {
 }
 
 data "coder_parameter" "instance_type" {
-  name        = "Instance Type"
-  description = "What instance type should your workspace use?"
-  default     = "t3.micro"
-  mutable     = false
+  name         = "instance_type"
+  display_name = "Instance Type"
```
Comment (bpmct) [inline examples/templates/aws-linux/main.tf:111]:
```suggestion
  display_name = "Instance type"
```

File: examples/templates/azure-linux/main.tf [hunk 7]
```diff
@@ -150,11 +151,12 @@ data "coder_parameter" "location" {
 }
 
 data "coder_parameter" "instance_type" {
-  name        = "Instance Type"
-  description = "What instance type should your workspace use?"
-  default     = "Standard_B4ms"
-  icon        = "/icon/azure.png"
-  mutable     = false
+  name         = "instance_type"
+  display_name = "Instance Type"
```
Comment (bpmct) [inline examples/templates/azure-linux/main.tf:155]:
```suggestion
  display_name = "Instance type"
```

File: examples/templates/azure-linux/main.tf [hunk 8]
```diff
@@ -202,12 +204,13 @@ data "coder_parameter" "instance_type" {
...
 
 data "coder_parameter" "home_size" {
-  name        = "Home Volume Size"
-  description = "How large would you like your home volume to be (in GB)?"
-  default     = 20
-  type        = "number"
-  icon        = "/icon/azure.png"
-  mutable     = false
+  name         = "home_size"
+  display_name = "Home Volume Size"
```
Comment (bpmct) [inline examples/templates/azure-linux/main.tf:208]:
```suggestion
  display_name = "Home volume size"
```

File: examples/templates/do-linux/main.tf [hunk 9]
```diff
@@ -52,10 +52,12 @@ variable "step2_do_admin_ssh_key" {
 }
 
 data "coder_parameter" "droplet_image" {
-  name    = "Which Droplet image would you like to use for your workspace?"
-  default = "ubuntu-22-04-x64"
-  type    = "string"
-  mutable = false
+  name         = "droplet_image"
+  display_name = "Droplet Image"
```
Comment (bpmct) [inline examples/templates/do-linux/main.tf:56]:
```suggestion
  display_name = "Droplet image"
```

File: examples/templates/do-linux/main.tf [hunk 10]
```diff
@@ -109,11 +111,13 @@ data "coder_parameter" "droplet_image" {
 }
 
 data "coder_parameter" "droplet_size" {
-  name    = "Which Droplet configuration would you like to use?"
-  default = "s-1vcpu-1gb"
-  type    = "string"
-  icon    = "/icon/memory.svg"
-  mutable = false
+  name         = "droplet_size"
+  display_name = "Droplet Size"
```
Comment (bpmct) [inline examples/templates/do-linux/main.tf:115]:
```suggestion
  display_name = "Droplet size"
```

**Review by bpmct — APPROVED** [review]
Can you change all casing to title case?

e.g.
- `Docker Image` -> `Docker image`
- `Home Volume Size` -> `Home volume size`

**Review by mtojek — APPROVED** [review]

**Review by bpmct — APPROVED** [review]

**Pull request merged**